crates/worker/src/partition/state_machine/command_interpreter/mod.rs

```diff
@@ -1203,7 +1188,13 @@ where
 
                     // Load state and write completion
                     let value = state
-                        .load_state(&invocation_metadata.service_id, &key)
+                        .load_state(
+                            &invocation_metadata
+                                .invocation_target
+                                .as_keyed_service_id()
+                                .unwrap(),
```

Comment:
I guess the way to implement this a bit more defensively would be to only load state if `as_keyed_service_id` returns `Some` and otherwise complete the `GetState` entry with an error. The upside would be that we wouldn't panic in case a wrong journal entry is being sent by a SDK and our filter fails.

Reply:
> I guess the way to implement this a bit more defensively would be to only load state if as_keyed_service_id returns Some and otherwise complete the GetState entry with an error.

I'm not sure I agree with completing `GetState` with an error, because this case happens only in case there is a runtime bug (that is, the invoker sent an invalid entry for given service/handler type). Perhaps how about i just close the PP with an error by propagating an error here?